### PR TITLE
[#113][bugfix] Study Room Deletion and Chat Bugs

### PR DESCRIPTION
**Pull request by mnashat**
This issue is concerned with the following Study Room bugs:
- sending chat messages fails
- deleting a study room fails

Causes:
- sending chat messages make a request with a user email that is wrapped in double quotes, causing database related backend function to not find the user in the database
- deleting a study room sends a request with incorrect type (`post` request instead of a `delete` request), incorrect body structure (body was structured for `post` requests, not `delete` requests), and to the incorrect route. These issues seem to have arisen when the backend was refactored, changing the expected requests by the backend.

Solutions:
- modify chat message requests to send the `email` param without the wrapping double quotes. Subsequently, this also required a small modification to how fetched chat messages are read in the frontend when differentiating between logged in user's messages and other participant messages.
- modify request type, body/API call structure, and route to reflect the expected data by the refactored backend

Resolution: Resolved

### CODE REVIEW

File: src/components/StudyRoom/StudyRoomChat/ChatFooter.js [hunk 1]
```diff
@@ -7,8 +7,8 @@ export const ChatFooter = () => {
     const handleSendMessage = (e) => {
         const studyRoomId = window.location.href.split("/")[window.location.href.split("/").length - 1];
         e.preventDefault();
-        const email = localStorage.getItem('email')
-        axios.post(`${process.env.REACT_APP_BASE_URL}message/send`, {studyRoomID: studyRoomId, email    : email, content: message})
+        const email = JSON.parse(localStorage.getItem('email'))
```
Comment (kikkiii21) [inline src/components/StudyRoom/StudyRoomChat/ChatFooter.js:10]: this should use the const email = GetAuthentication().email;

File: src/components/StudyRoom/StudyRoomChat/ChatFooter.js [hunk 2]
```diff
@@ -7,8 +7,8 @@ export const ChatFooter = () => {
...
+        axios.post(`${process.env.REACT_APP_BASE_URL}message/send`, {studyRoomID: studyRoomId, email: email, content: message})
             .then((res) => {console.log(res)})
```
Comment (kikkiii21) [inline src/components/StudyRoom/StudyRoomChat/ChatFooter.js:12]: do we need the console log?

**Review by kikkiii21 — APPROVED** [review]

File: src/components/StudyRoom/StudyRoomChat/ChatFooter.js [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
 
     const handleSendMessage = (e) => {
-        const studyRoomId = window.location.href.split("/")[window.location.href.split("/").length - 1];
         e.preventDefault();
-        const email = localStorage.getItem('email')
-        axios.post(`${process.env.REACT_APP_BASE_URL}message/send`, {studyRoomID: studyRoomId, email    : email, content: message})
-            .then((res) => {console.log(res)})
-        setMessage('');
+        const studyRoomId = window.location.href.split("/")[window.location.href.split("/").length - 1];
+        const email = GetAuthentication().email;
```
Comment (guaj) [inline src/components/StudyRoom/StudyRoomChat/ChatFooter.js:11]: const { email } = GetAuthentification();

**Review by guaj — APPROVED** [review]

File: src/components/StudyRoom/StudyRoomChat/ChatFooter.js (hunk 3 above)
Comment (mnashat) [reply src/components/StudyRoom/StudyRoomChat/ChatFooter.js:11]: Why should we get all of the attributes? Only email is being used in this case.

**Pull request merged**